src/index.ts

```diff
@@ -25,13 +25,13 @@ function doCallback() {
   });
 }
 
-export function useSubscribe<Tx, D, R extends D>(
-  r: Subscribable<Tx, D> | null | undefined,
-  query: (tx: Tx) => Promise<R>,
-  def: R,
+export function useSubscribe<Tx, Data, QueryRet extends Data, Default>(
+  r: Subscribable<Tx, Data> | null | undefined,
+  query: (tx: Tx) => Promise<QueryRet>,
+  def: Default,
   deps: Array<unknown> = [],
-): R {
-  const [snapshot, setSnapshot] = useState<R>(def);
+) {
+  const [snapshot, setSnapshot] = useState<QueryRet | undefined>(undefined);
```

Comment:
I find it easier to reason about this by setting the state to `def` here.

But both should behave the same so your call here.